Add methods to hilo game

zoejoblin opened the pull request.

Commit:
create new game class and define methods

ariel-reyes-ow commented on new_game.rb:
```diff
@@ -0,0 +1,41 @@
+class NewGame
+    @input
```
Instead of using an instance variable, there is also a convention we follow where we use the object initialization pattern:`https://ruby-doc.org/docs/ruby-doc-bundle/UsersGuide/rg/objinitialization.html`.  It makes the code more readable as it using set at the first bit.

zoejoblin replied:
I considered doing that but thought it might be good ux to have an explicit start method the user could call, but if an init method makes more sense I'll change it!

zoejoblin replied:
Cool - I hav refactored to include an initialize method and a start method - let me know your thoughts :)

Commit:
add initialize method

ariel-reyes-ow commented on hilo.rb:
```diff
@@ -1,20 +1,6 @@
-puts (guess == random_number) ? "Correct! You win" : "Sorry, you ran out of guesses. The number was #{random_number}"
+require_relative 'new_game'
+
+new_game = NewGame.new
```
Just curious why you opted to have the `NewGame` class in a different file.  On the side there is also an option to have multiple classes in one file: https://docs.ruby-lang.org/en/3.0/syntax/modules_and_classes_rdoc.html.  

Also, why not use `private methods` instead?  Just something to think about.

zoejoblin replied:
I started with it in the same file but thought it was tidier to separate it out from thee running file. Happy to change if it makes more sense to keep them in the same file?

zoejoblin replied:
Yep makes sense to use private methods - will update

zoejoblin replied:
Would you suggest using a class or a module for this game? I don't fully understand the difference after reading that documentation

Commit:
fix logic to set chances variable in order to display correct value to user

ariel-reyes-ow commented on new_game.rb:
```diff
@@ -0,0 +1,42 @@
+        end
+    end
+
+    def is_correct
```
Normally when its a boolean method we end it with a `?`

ariel-reyes-ow commented on new_game.rb:
```diff
@@ -0,0 +1,42 @@
+    end 
+
+    def check_guess
+        response = nil
```
Ruby is also a bit cleve whereby it returns the method result. For example here you don't actually need a `response` instance variable.

zoejoblin replied:
Cool makes sense - although I want to return nil when there is no response in order to use my ternery operator on line 17 - is there a better way to do this?

ariel-reyes-ow commented on hilo.rb:
```diff
@@ -1,20 +1,6 @@
+
+random_number_range = [1, 100]
+no_of_chances = 6
+new_game = NewGame.new(random_number_range, no_of_chances)
```
Pardon for the nitpick, but would you opt for passing already the `random_number` so you don't have two params and get the position values in L6?

zoejoblin replied:
I don't quite understand what you mean sorry - can you give me an example?

Commit:
put all code in same file and make class methods priavte

Commit:
Refactor hilo game following group code review

Pass min and max values for random number range rather than a range
array. Fix indentation and spacing to mkae code more readable. Add some
todo's for furture refactoring.
